packages/expo-video/android/src/main/java/expo/modules/video/VideoModule.kt

```diff
@@ -0,0 +1,145 @@
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
+import kotlinx.coroutines.launch
+
+// https://developer.android.com/guide/topics/media/media3/getting-started/migration-guide#improvements_in_media3
+@androidx.annotation.OptIn(androidx.media3.common.util.UnstableApi::class)
+class VideoModule : Module() {
+  private val activity: Activity
+    get() = appContext.activityProvider?.currentActivity ?: throw Exceptions.MissingActivity()
+
+  override fun definition() = ModuleDefinition {
+    Name("ExpoVideo")
+
+    View(VideoView::class) {
+      Prop("player") { view: VideoView, player: VideoPlayer ->
+        view.videoPlayer = player
+        player.prepare()
+      }
+
+      Prop("nativeControls") { view: VideoView, useNativeControls: Boolean ->
+        view.playerView.useController = useNativeControls
+      }
+
+      Prop("contentFit") { view: VideoView, contentFit: ContentFit ->
+        view.playerView.resizeMode = contentFit.toResizeMode()
+      }
+
+      Prop("requiresLinearPlayback") { view: VideoView, requiresLinearPlayback: Boolean? ->
+        val linearPlayback = requiresLinearPlayback ?: false
+        view.playerView.setShowFastForwardButton(!linearPlayback)
+        view.playerView.setShowRewindButton(!linearPlayback)
+        view.playerView.setShowPreviousButton(!linearPlayback)
+        view.playerView.setShowNextButton(!linearPlayback)
+
+        // TODO: Make the requiresLinearPlayback hide only the scrubber instead of the whole progress bar. Maybe use custom layout for the player as the scrubber is not available?
+        val progressBar = view.playerView.findViewById<View>(androidx.media3.ui.R.id.exo_progress)
+        if (progressBar is DefaultTimeBar) {
+          progressBar.visibility = if (linearPlayback) {
+            View.GONE
+          } else {
+            View.VISIBLE
+          }
+        }
+        view.videoPlayer?.requiresLinearPlayback = linearPlayback
+      }
+    }
+
+    Class(VideoPlayer::class) {
+      Constructor { source: String ->
+        val mediaItem = MediaItem.fromUri(source)
+        VideoPlayer(activity.applicationContext, mediaItem)
+      }
+
+      Property("isPlaying")
+        .get { ref: VideoPlayer ->
+          ref.isPlaying
+        }
+
+      Property("isLoading")
+        .get { ref: VideoPlayer ->
+          ref.isLoading
+        }
+
+      Property("isMuted")
+        .get { ref: VideoPlayer ->
+          ref.isMuted
+        }
+        .set { ref: VideoPlayer, isMuted: Boolean ->
+          appContext.mainQueue.launch {
+            ref.isMuted = isMuted
+          }
+        }
+
+      Property("volume")
+        .get { ref: VideoPlayer ->
+          ref.volume
+        }
+        .set { ref: VideoPlayer, volume: Float ->
+          appContext.mainQueue.launch {
+            ref.userVolume = volume
+            ref.volume = volume
+          }
+        }
+
+      AsyncFunction("getPlaybackSpeed") { ref: VideoPlayer ->
```

Comment:
It seems a bit odd if we have an async function for getter and sync function for setter 🤔 Is running it on the main thread necessary?

Reply:
Seems like a leftover after some of Kamil's tests that I missed 😅 , should be ok now 😄